src/AutoRest.CSharp/Common/Output/Models/Types/BuildContext.cs

```diff
@@ -9,20 +9,25 @@ namespace AutoRest.CSharp.Output.Models.Types
 {
     internal class BuildContext
     {
-        public BuildContext(CodeModel codeModel, SourceInputModel? sourceInputModel)
+        public BuildContext(CodeModel codeModel, SourceInputModel? sourceInputModel) : this(codeModel, sourceInputModel, Configuration.LibraryName, Configuration.Namespace)
+        { }
+
+        public BuildContext(CodeModel codeModel, SourceInputModel? sourceInputModel, string defaultLibraryName, string defaultNamespace)
         {
             CodeModel = codeModel;
             SchemaUsageProvider = new SchemaUsageProvider(codeModel);
             SourceInputModel = sourceInputModel;
+            DefaultLibraryName = defaultLibraryName;
+            DefaultNamespace = defaultNamespace;
         }
 
         public OutputLibrary? BaseLibrary { get; protected set; }
 
         public CodeModel CodeModel { get; }
         public SchemaUsageProvider SchemaUsageProvider { get; }
         public string DefaultName => CodeModel.Language.Default.Name;
-        public string DefaultNamespace => Configuration.Namespace;
-        public string DefaultLibraryName => Configuration.LibraryName;
+        public string DefaultNamespace { get; }
```

Comment:
It will decouple the `DefaultName` and `DefaultNamespace` in BuildContext from the Configuration. When the value in Configuration is changed, the BuildContext will not aware it.

Need to investigate if the changes in Configuration is needed to reflect the BuildContext or not.

Reply:
The preview PR doesn't include any code change. I think we're fine here.